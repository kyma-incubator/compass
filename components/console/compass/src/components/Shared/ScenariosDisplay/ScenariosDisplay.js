import React from 'react';
import PropTypes from 'prop-types';
import { Badge } from 'fundamental-react';
import { EMPTY_TEXT_PLACEHOLDER } from '../../../shared/constants';

ScenariosDisplay.propTypes = {
  scenarios: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  className: PropTypes.string,
  emptyPlaceholder: PropTypes.string,
};

ScenariosDisplay.defaultProps = {
  emptyPlaceholder: EMPTY_TEXT_PLACEHOLDER,
};

export default function ScenariosDisplay({
  scenarios,
  className,
  emptyPlaceholder,
}) {
  if (!scenarios.length) {
    return <span className={className}>{emptyPlaceholder}</span>;
  }

  return (
    <span className={className}>
      {scenarios.map(scenario => (
        <Badge
          key={scenario}
          className="fd-has-margin-right-tiny fd-has-background-color-background-1 fd-has-color-text-3"
          modifier="filled"
        >
          {scenario}
        </Badge>
      ))}
    </span>
  );
}
