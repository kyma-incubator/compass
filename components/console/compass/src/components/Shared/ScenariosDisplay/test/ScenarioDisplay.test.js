import React from 'react';
import renderer from 'react-test-renderer';
import ScenariosDisplay from '../ScenariosDisplay';

describe('ScenariosDisplay', () => {
  // for "Warning: componentWillReceiveProps has been renamed"
  console.error = jest.fn();
  console.warn = jest.fn();

  afterEach(() => {
    console.error.mockReset();
    console.warn.mockReset();
  });

  afterAll(() => {
    expect(console.error.mock.calls[0][0]).toMatchSnapshot();
    if (console.warn.mock.calls.length) {
      expect(console.warn.mock.calls[0][0]).toMatchSnapshot();
    }
  });

  it('Renders scenario list', () => {
    const component = renderer.create(
      <ScenariosDisplay scenarios={['a', 'b', 'c']} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders default placeholder when there are no scenarios', () => {
    const component = renderer.create(<ScenariosDisplay scenarios={[]} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders custom placeholder when there are no scenarios', () => {
    const component = renderer.create(
      <ScenariosDisplay scenarios={[]} emptyPlaceholder={'No scenarios'} />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders custom class name', () => {
    const component = renderer.create(
      <ScenariosDisplay scenarios={['a', 'b', 'c']} className="class-name" />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
