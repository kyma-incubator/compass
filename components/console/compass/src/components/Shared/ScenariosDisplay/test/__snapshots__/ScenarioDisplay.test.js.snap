// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScenariosDisplay Renders custom class name 1`] = `
<span
  className="class-name"
>
  <span
    className="fd-badge fd-badge--filled fd-has-margin-right-tiny fd-has-background-color-background-1 fd-has-color-text-3"
  >
    a
  </span>
  <span
    className="fd-badge fd-badge--filled fd-has-margin-right-tiny fd-has-background-color-background-1 fd-has-color-text-3"
  >
    b
  </span>
  <span
    className="fd-badge fd-badge--filled fd-has-margin-right-tiny fd-has-background-color-background-1 fd-has-color-text-3"
  >
    c
  </span>
</span>
`;

exports[`ScenariosDisplay Renders custom placeholder when there are no scenarios 1`] = `
<span>
  No scenarios
</span>
`;

exports[`ScenariosDisplay Renders default placeholder when there are no scenarios 1`] = `
<span>
  -
</span>
`;

exports[`ScenariosDisplay Renders scenario list 1`] = `
<span>
  <span
    className="fd-badge fd-badge--filled fd-has-margin-right-tiny fd-has-background-color-background-1 fd-has-color-text-3"
  >
    a
  </span>
  <span
    className="fd-badge fd-badge--filled fd-has-margin-right-tiny fd-has-background-color-background-1 fd-has-color-text-3"
  >
    b
  </span>
  <span
    className="fd-badge fd-badge--filled fd-has-margin-right-tiny fd-has-background-color-background-1 fd-has-color-text-3"
  >
    c
  </span>
</span>
`;
